finance_dl/amazon.py

```diff
@@ -281,9 +293,16 @@ def get_source():
             if order_id not in page_source:
                 raise ValueError('Failed to retrieve information for order %r'
                                  % (order_id, ))
+            m = re.search('(?:Digital Order: |Order Placed: *\n *</b>\n) *[A-Za-z]* \d+, (\d{4})',
+                         page_source)
+            if m is None:
+                raise ValueError('Failed to get date for order %r' % order_id)
+            invoice_path = self.get_invoice_path(m[1], order_id)
```

Comment:
This introduces an additional assumption needed for the downloader to complete successfully. I imagine the vast majority of invoices will parse fine, but the one that will require manual intervention whereas in current HEAD it wouldn't.
All in all, given that users can choose to not enable this option, this seems fine.

Reply:
Good point, I've made this not be an error if `dir_per_year` is off.